### PR TITLE
use ibc v4.3.0

### PR DESCRIPTION
**Pull request by faddat**
# Description

Separated from #805.  Equally important.

fix: do we want another cosmos hub quicksilver ica incident?

Since we do not want that, let's always ship with the latest available code.

PS: yes I grouped the direct imports.

PPS: no, that is not a refactor, if they are separated there should be a comment explaining grouping.

## Type of change

If you've checked more than one of the first three boxes, consider splitting this PR into multiple PRs!

- [ ] `Feature`: Changes and/or adds code behavior, irrelevant to bug fixes
- [x] `Fix`: Changes and/or adds code behavior, specifically to fix a bug
- [ ] `Refactor`: Changes existing code style, naming, structure, etc.
- [ ] `Testing`: Adds testing
- [ ] `Docs`: Adds documentation

## Regression tests

If `Refactor`, describe the new or existing tests that verify no behavior was changed or added where refactors were introduced.

## New behavior tests

If `Feature` or `Fix`, describe the new or existing tests that verify the new behavior is correct and expected.

## Versioning Implications

- [x] This PR will affect [semantic versioning as defined for ICS](../CONTRIBUTING.md#semantic-versioning)

If the above box is checked, which version should be bumped?

- [x] `MAJOR`: Consensus breaking changes to both the provider and consumers(s), including updates/breaking changes to IBC communication between provider and consumer(s)
- [ ] `MINOR`: Consensus breaking changes which affect either only the provider or only the consumer(s)
- [ ] `PATCH`: Non consensus breaking changes

## Targeting

Please select one of the following:

- [x] This PR is only relevant to main
- [ ] This PR is relevant to main, and should also be back-ported to ____ (ex: v1.0.0 and v1.1.0)
- [ ] This PR is only relevant to ____ (ex: v1.0.0, v1.1.0, and v1.2.0)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,13 +3,18 @@ module github.com/cosmos/interchain-security
 go 1.18
 
 require (
-	github.com/cosmos/cosmos-sdk v0.45.11
+	github.com/confio/ics23/go v0.9.0
+	github.com/cosmos/cosmos-sdk v0.45.12
```
Comment (jtremback) [inline go.mod:7]: Note for tomorrow- does this need to be 45.13?

File: go.mod [hunk 2]
```diff
@@ -18,6 +23,7 @@ require (
 	github.com/tendermint/tm-db v0.6.7
 	github.com/tidwall/gjson v1.14.4
 	golang.org/x/crypto v0.5.0 // indirect
+	golang.org/x/exp v0.0.0-20220722155223-a9213eeb770e
```
Comment (jtremback) [inline go.mod:26]: What's this one?

**Review by jtremback — APPROVED** [review]
LGTM

File: go.mod (hunk 1 above)
Comment (jtremback) [reply go.mod:7]: We override it in the replace block with 45.13 anyway

**Review by smarshall-spitzbart — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (faddat) [reply go.mod:7]: Case it should say 13 but there's a reason why it was bundled together with the one that upgrades to 4515

**Comment by faddat** [comment]
I fix

**Comment by faddat** [comment]
thought this was approved?

**Pull request merged**